common/persistence/pinot/pinotVisibilityStore.go

```diff
@@ -770,20 +797,21 @@ func (v *pinotVisibilityStore) getListWorkflowExecutionsByQueryQuery(tableName s
 	// if customized query is empty, directly return
 	if requestQuery == "" {
 		query.addOffsetAndLimits(token.From, request.PageSize)
-		return query.String()
+		return query.String(), nil
 	}
 
 	requestQuery = filterPrefix(requestQuery)
 	if common.IsJustOrderByClause(requestQuery) {
 		query.concatSorter(requestQuery)
 		query.addOffsetAndLimits(token.From, request.PageSize)
-		return query.String()
+		return query.String(), nil
 	}
 
 	comparExpr, orderBy := parseOrderBy(requestQuery)
 	comparExpr, err = v.pinotQueryValidator.ValidateQuery(comparExpr)
 	if err != nil {
-		v.logger.Error(fmt.Sprintf("pinot query validator error: %s", err))
+		v.logger.Error(fmt.Sprintf("pinot query validator error: %s, query: %s", err, request.Query))
+		return "", err
```

Comment:
```suggestion
		return "", fmt.Errorf("query validation: %w, query %s", err, request.Query)
```

Then you can avoid logging inside a component and only log failure higher by the stack.

Reply:
Read more on https://github.com/uber-go/guide/blob/master/style.md#error-wrapping and https://github.com/uber-go/guide/blob/master/style.md#handle-errors-once

Reply:
Thanks, great to know these, will update them!